static/app/views/settings/components/forms/selectField.tsx

```diff
@@ -79,7 +84,43 @@ export default class SelectField<
   };
 
   render() {
-    const {multiple, allowClear, small, ...otherProps} = this.props;
+    const {allowClear, confirm, multiple, small, ...otherProps} = this.props;
+
+    if (confirm) {
+      return (
+        <InputField
+          {...otherProps}
+          alignRight={small}
+          field={({onChange, onBlur, required: _required, ...props}) => (
+            <Confirm
+              message={confirm.message}
```

Comment:
I tried to copy `confirm` from `BooleanField`, but I can't figure out how to modulate the `message` with the new value.